### PR TITLE
Remove proxygen dependency and use cpp-httplib instead

### PR DESCRIPTION
**Pull request by gtrepta**
https://github.com/runtimeverification/firefly/issues/243

Sister PR: https://github.com/kframework/evm-semantics/pull/656

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[submodule "deps/cpp-httplib"]
+	path = deps/cpp-httplib
+	url = https://github.com/yhirose/cpp-httplib.git
+    ignore = untracked
```
Comment (ehildenb) [inline .gitmodules]: Something weird going on here formatting wise.

Comment (gtrepta) [reply .gitmodules]: Does `.gitmodules` require tab indentation?

File: client-c/main.cpp [hunk 2]
```diff
@@ -256,3 +218,58 @@ void openSocket() {
...
+      }
+      case '[':{
+        bracket_counter_++;
+        break;
+      }
+      case ']':{
+        bracket_counter_--;
+        break;
+      }
+    }
```
Comment (ehildenb) [inline client-c/main.cpp]: This loop can be factored out into its own function which both `countBrackets` and `doneReading` can use I think?

Comment (gtrepta) [reply client-c/main.cpp]: Yeah, I could factor out part of the loop body.

File: .gitmodules (hunk 1 above)
Comment (ehildenb) [reply .gitmodules]: No, but I guess I'd like it to be consistent.

**Pull request merged**